ast/format.go

```diff
@@ -0,0 +1,356 @@
+	sep := '\n'
+	for i, c := range n.Body {
+		f.formatNode(c)
+		if i < len(n.Body)-1 {
+			f.writeRune(sep)
+		}
+	}
+}
+
+func (f *formatter) formatBlockStatement(n *BlockStatement) {
+	sep := '\n'
+	for i, c := range n.Body {
+		f.formatNode(c)
+		if i < len(n.Body)-1 {
+			f.writeRune(sep)
+		}
+	}
+}
+
+func (f *formatter) formatExpressionStatement(n *ExpressionStatement) {
+	f.formatNode(n.Expression)
+}
+
+func (f *formatter) formatReturnStatement(n *ReturnStatement) {
+	f.writeString("return ")
+	f.formatNode(n.Argument)
+}
+
+func (f *formatter) formatOptionStatement(n *OptionStatement) {
+	f.writeString("option ")
+	f.formatNode(n.Declaration)
+}
+
+func (f *formatter) formatVariableDeclaration(n *VariableDeclaration) {
+	sep := ' '
+	for i, c := range n.Declarations {
+		f.formatNode(c)
+		if i < len(n.Declarations)-1 {
+			f.writeRune(sep)
+		}
+	}
+}
+
+func (f *formatter) formatVariableDeclarator(n *VariableDeclarator) {
+	f.formatNode(n.ID)
+	f.writeRune('=')
+	f.formatNode(n.Init)
+}
+
+func (f *formatter) formatArrayExpression(n *ArrayExpression) {
+	f.writeRune('[')
+
+	sep := ','
+	for i, c := range n.Elements {
+		f.formatNode(c)
+		if i < len(n.Elements)-1 {
+			f.writeRune(sep)
+		}
+	}
+
+	f.writeRune(']')
+}
+
+func (f *formatter) formatArrowFunctionExpression(n *ArrowFunctionExpression) {
+	f.writeRune('(')
+
+	sep := ','
+	for i, c := range n.Params {
+		// treat properties differently than in general case
+		f.formatArrowFunctionArgument(c)
+		if i < len(n.Params)-1 {
+			f.writeRune(sep)
+		}
+	}
+
+	f.writeString(")=>")
+	f.formatNode(n.Body)
+}
+
+func (f *formatter) formatUnaryExpression(n *UnaryExpression) {
+	f.writeString(n.Operator.String())
+	f.formatNode(n.Argument)
+}
+
+func (f *formatter) formatBinaryExpression(n *BinaryExpression) {
+	f.formatNode(n.Left)
+	f.writeString(n.Operator.String())
+	f.formatNode(n.Right)
+}
+
+func (f *formatter) formatLogicalExpression(n *LogicalExpression) {
+	f.formatNode(n.Left)
+	f.writeString(n.Operator.String())
+	f.formatNode(n.Right)
+}
+
+func (f *formatter) formatCallExpression(n *CallExpression) {
+	f.formatNode(n.Callee)
+	f.writeRune('(')
+
+	sep := ','
+	for i, c := range n.Arguments {
+		// treat ObjectExpression as argument in a special way
+		// (an object as argument doesn't need braces)
+		if oe, ok := c.(*ObjectExpression); ok {
+			f.formatObjectExpressionAsFunctionArgument(oe)
+		} else {
+			f.formatNode(c)
+		}
+
+		if i < len(n.Arguments)-1 {
+			f.writeRune(sep)
+		}
+	}
+
+	f.writeRune(')')
+}
+
+func (f *formatter) formatPipeExpression(n *PipeExpression) {
+	f.formatNode(n.Argument)
+	f.writeString("|>")
+	f.formatNode(n.Call)
+}
+
+func (f *formatter) formatConditionalExpression(n *ConditionalExpression) {
+	f.formatNode(n.Test)
+	f.writeRune('?')
+	f.formatNode(n.Consequent)
+	f.writeRune(':')
+	f.formatNode(n.Alternate)
+}
+
+func (f *formatter) formatMemberExpression(n *MemberExpression) {
+	f.formatNode(n.Object)
+	f.writeRune('.')
```

Comment:
This needs to check if the Property is an Identifier or a StringLiteral. If it a string literal you need to output the `["this is a property"]` format.